packages/legend-query-builder/src/application/QueryBuilderSetting.ts

```diff
@@ -18,10 +18,14 @@ import type { SettingConfigurationData } from '@finos/legend-application';
 
 export enum QUERY_BUILDER_SETTING_KEY {
   SHOW_POST_FILTER_PANEL = 'query-builder.showPostFilterPanel',
+  RECENTLY_VIEWED_QUERIES = 'query-builder.recentlyViewedQueries',
 }
 
 export const QUERY_BUILDER_SETTING_CONFIG: SettingConfigurationData = {
   [QUERY_BUILDER_SETTING_KEY.SHOW_POST_FILTER_PANEL]: {
     defaultValue: false,
   },
+  [QUERY_BUILDER_SETTING_KEY.RECENTLY_VIEWED_QUERIES]: {
```

Comment:
You're confused here about the purpose of the setting store, this is not part of setting, this is another store you need (something like user state, which is very very customized based on the application)